cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -389,6 +389,52 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         }
     });
 
+    SelfPacedDueDateEditor = AbstractEditor.extend({
+        fieldName: 'due_num_weeks',
+        templateName: 'self-paced-due-date-editor',
+        className: 'modal-section-content has-actions due-date-input grading-due-date',
+
+        events: {
+            'click .clear-date': 'clearValue',
+            'keyup #due_in': 'validateDueIn'
+        },
+
+        getValue: function() {
+            return parseInt(this.$('#due_in').val());
+        },
+
+        validateDueIn: function() {
+            if (parseInt(this.$('#due_in').val()) > 18){
+                this.$('#warning').show();
+                BaseModal.prototype.disableActionButton.call(this.parent, 'save');
+            }
+            else {
+                this.$('#warning').hide();
+                BaseModal.prototype.enableActionButton.call(this.parent, 'save');
+            }
+        },
+
+        clearValue: function(event) {
+            event.preventDefault();
+            this.$('#due_in').val('');
+        },
+
+        afterRender: function() {
+            AbstractEditor.prototype.afterRender.call(this);
+            this.$('.field-due-in input').val(this.model.get('due_num_weeks'));
+        },
+
+        getRequestData: function() {
+            if (this.getValue() < 18) {
```

Comment:
I wonder if this is necessary on top of your `validateDueIn` function. Likely can't hurt. Do you know what happens if it hits this line and it's false? Would the requestData be `undefined`? Do we handle that unlikely case?

Reply:
If it's false, the requestData is `None` and it doesn't get saved,  but I should handle if it is `undefined`! I put this extra condition in because although I disable the save button if the input is greater than 18, there is a really short split second where I could still press save (if the user really really tries to be quick).

Reply:
The `None` / `undefined` cases might actually be the same.